src/main/scala/com/github/unchama/seichiassist/subsystems/gachaprize/GachaPrizeAPI.scala

```diff
@@ -0,0 +1,173 @@
+package com.github.unchama.seichiassist.subsystems.gachaprize
+
+import cats.Monad
+import com.github.unchama.seichiassist.subsystems.gachaprize.domain.gachaevent.{
+  GachaEvent,
+  GachaEventName
+}
+import com.github.unchama.seichiassist.subsystems.gachaprize.domain.gachaprize.{
+  GachaPrize,
+  GachaPrizeId
+}
+import com.github.unchama.seichiassist.subsystems.gachaprize.domain.{
+  CanBeSignedAsGachaPrize,
+  StaticGachaPrizeFactory
+}
+
+trait GachaEventReadAPI[F[_]] {
+
+  import cats.implicits._
+
+  protected implicit val F: Monad[F]
+
+  /**
+   * @return 現在作成されているガチャイベントの一覧を取得する作用
+   */
+  def createdGachaEvents: F[Vector[GachaEvent]]
+
+  /**
+   * @return 指定された`gachaEventName`のイベントが存在すれば返す作用
+   */
+  private def findGachaEvent(gachaEventName: GachaEventName): F[Option[GachaEvent]] = for {
+    gachaEvents <- createdGachaEvents
+  } yield gachaEvents.find(_.eventName == gachaEventName)
+
+  /**
+   * 指定された[[GachaEventName]]のイベントが存在するか調べる
+   * @return 存在すれば`true`、存在しないなら`false`を返す作用
+   */
+  final def isExistsGachaEvent(gachaEventName: GachaEventName): F[Boolean] = for {
```

Comment:
```suggestion
  final def existsGachaEvent(gachaEventName: GachaEventName): F[Boolean] = for {
```

cf. https://qiita.com/yskszk/items/5a7f99c974773f03a82a